src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -90,104 +109,110 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
+			mostRecentHistory = completedHistories.get(0);
+		} else {
+			// There are 2 normal states where there are no completed histories
+			// 1. We upgraded to the new standardization code and there are no completed standardizations
+			// 2. This is a new install and there are no completed standardizations
+			mostRecentHistory = new StandardizationHistory();
+			StandardizationSettingsConfigCheckResponseDTO checkStandarizerOutput = chemStructureService.checkStandardizerSettings(mostRecentHistory, currentRawStandardizerSettings);
+			if(Parent.countParents() == 0) {
+				// We need to call checkStandardization settings as it is responsible for making sure the applied settings are
+				// filled in approriatly before we save the history with applied settings.
+				if(!checkStandarizerOutput.getValid()) {
+					// If the settings are invalid then we don't save a standardization history as there has never been a valid state
+					String initialSettingsInvalidReasons = checkStandarizerOutput.getInvalidReasons().stream().collect(Collectors.joining(System.lineSeparator()));
+					logger.info("Initial standardization settings are invalid: " + initialSettingsInvalidReasons);
 				} else {
-					logger.warn(
-							"Standardization is in an unknown state because there are no rows in standardization history table.");
-					if (currentStandardizationSettings.getShouldStandardize() == false) {
-						logger.info(
-								"Standardization is turned off so marking the database as not requiring standardization at this time");
-						standardizationSettings.setNeedsStandardization(false);
-						standardizationSettings.persist();
-					} else {
-						logger.warn(
-								"Standardization is turned on so we don't know the current database standardization state. Assuming no standardization is needed.");
-						standardizationSettings.setNeedsStandardization(false);
-						standardizationSettings.persist();
-					}
+
+					// If there is no standardization history the lets create one
+					mostRecentHistory = StandardizationDryRunCompound.addStatsToHistory(mostRecentHistory);
+					mostRecentHistory.setStructuresStandardizedCount(0);
+					mostRecentHistory.setStructuresUpdatedCount(0);
+					mostRecentHistory.setStandardizationComplete(new Date());
+					mostRecentHistory.setStandardizationStatus("complete");
+					mostRecentHistory.setStandardizationUser("acas");
+					mostRecentHistory.setStandardizationReason("Initial standardization record");
+					mostRecentHistory.persist();
+
 				}
 			}
-		} catch (StandardizerException e) {
-			logger.error("Caught error on checking standardization state", e);
+
+		}
+
+		// Do a config check to verify that settings are valid and if we need to restandardize or not
+		StandardizationSettingsConfigCheckResponseDTO configCheckResponse = chemStructureService.checkStandardizerSettings(mostRecentHistory, currentRawStandardizerSettings);
+
+		// We should only ever really have one standardization settings row at any given
```

Comment:
I think it is worth explaining here what a "StandardizationSettings" is. A lot of these variables and classes have similar names. As I understand it `StandardizationSettings` is actually the current state of the system, specifically whether it needs restandardization and why. `StandardizationStatus` might be a better name, or `CurrentStandardizationStatus`

Reply:
Makes sense but unclear if you are suggesting that I refactor the table name, class name and variable names to match `CurrentStandardizationStatus ` as part of this PR or to just note it in this comment.

Reply:
I added some comments at least for now.

Reply:
I realize after writing this that the way to refactor this is to just drop the settings table and make settings just a DTO because there is no point in saving the row in a table if we are completely replacing it on every check.

Reply:
@brianbolt your last comment makes a lot of sense to me. I think it's worth doing as part of this PR, just to leave this code in a clean place. My hope is that we don't have a lot more work to do in this area of the code, so there may not be another good opportunity to make that change.